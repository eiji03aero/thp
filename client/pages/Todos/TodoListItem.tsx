import * as React from "react";
import * as cn from "classnames";
import styled from "styled-components";

import { Todo } from "./Todo";

const TodoListItemDiv = styled.div`
  display: flex;
  span {
    margin-left: 1rem;
  }

  .completed {
    text-decoration: line-through;
  }
`;

interface Props {
  todo: Todo;
  onToggleCompleted: (e: React.SyntheticEvent, id: string) => void;
}

export const TodoListItem: React.StatelessComponent<Props> = ({
  todo,
  onToggleCompleted,
}) => {
  return (
    <TodoListItemDiv className={cn({ 'completed': todo.completed })}>
      <input
        type="checkbox"
        value={todo.completed.toString()}
        onChange={(e: React.SyntheticEvent): void => onToggleCompleted(e, todo.id)}
      />
      <span>{ todo.title }</span>
    </TodoListItemDiv>
  );
};
