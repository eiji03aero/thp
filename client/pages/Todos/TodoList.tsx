import * as React from "react";
import * as _ from "lodash";
import styled from "styled-components";

import { TodoListItem } from "./TodoListItem";
import { Todo } from "./Todo";

const TodoListDiv = styled.div`
  display: block;
`;

interface Props {
  todos: Todo[];
  onToggleCompleted: (e: React.SyntheticEvent, id: string) => void;
}

export const TodoList: React.StatelessComponent<Props> = ({
  todos,
  onToggleCompleted,
}) => {
  return (
    <TodoListDiv>

      { _.map(todos, todo => (
        <TodoListItem
          todo={todo}
          onToggleCompleted={onToggleCompleted}
        />
      ))}

    </TodoListDiv>
  );
};
