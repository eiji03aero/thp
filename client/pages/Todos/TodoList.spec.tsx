import * as React from 'react';
import { shallow } from "enzyme";
import * as renderer from 'react-test-renderer';
import { TodoList } from "./TodoList";
import { TodoListItem } from "./TodoListItem";

describe('TodoList', () => {
  const todos = [ { id: '0', title: 'test', completed: false } ];

  it('snapshot', () => {
    const tree = renderer
      .create(<TodoList todos={todos} onToggleCompleted={e => {}}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render proper number of Items', () => {
    const wrapper = shallow(<TodoList todos={todos} onToggleCompleted={e => {}}/>);
    expect(wrapper.find(TodoListItem).length).toBe(1);
  });
});
