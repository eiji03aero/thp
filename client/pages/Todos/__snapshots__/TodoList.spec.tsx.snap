// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList snapshot 1`] = `
<div
  className="sc-bwzfXH gbieAY"
>
  <div
    className="sc-bdVaJa cvGKIk"
  >
    <input
      onChange={[Function]}
      type="checkbox"
      value="false"
    />
    <span>
      test
    </span>
  </div>
</div>
`;
